frontend/src/app/Settings/Users/ManageUsers.tsx

```diff
@@ -120,35 +115,103 @@ const ManageUsers: React.FC<Props> = ({
     updateIsUserEdited(false);
   };
 
-  const onHandleCreateNewUser = (newUserInvite: NewUserInvite) => {
-    // TODO: validate form
-    onCreateNewUser(newUserInvite);
+  const handleUpdateUser = (userId: string) => {
+    const selectedRoleDescription = usersState[userId]
+      .roleDescription as RoleDescription;
+    const selectedOrganizationRole = RoleDescriptionToOrgRole[
+      selectedRoleDescription
+    ] as OrganizationRole;
+    updateUserRole({
+      variables: {
+        id: userId,
+        role: selectedOrganizationRole,
+      },
+    })
+      .then(() => {
+        updateIsUserEdited(false);
 
-    const fullName = displayFullName(
-      newUserInvite.firstName,
-      null,
-      newUserInvite.lastName
-    );
+        const user = usersState[userId];
 
-    let successAlert = (
-      <Alert
-        type="success"
-        title={`Invitation sent to ${fullName}`}
-        body={`They will receive an invitation to create an account at the email address provided`}
-      />
-    );
+        const fullName = displayFullNameInOrder(
+          user.firstName,
+          user.middleName,
+          user.lastName
+        );
 
-    showNotification(toast, successAlert);
-    updateShowAddUserModal(false);
+        showNotification(
+          toast,
+          <Alert
+            type="success"
+            title="Changes Saved"
+            body={`${fullName}'s settings have been saved`}
+          />
+        );
+      })
+      .catch(setError);
```

Comment:
Oh I see, you wanted to get the error from the async operation back into the component. What happens if you don't do that?

Reply:
I saw this somewhere else in the app and figured it was reasonable to just blow up to the nearest error boundary if we hit an error here. The other option would be to use `showNotification` to show an error toast, but then I'd also have to wire in some custom error reporting. The error boundary was mostly out of a _bit_ of laziness to get the free error reporting from the error boundary component.